send_upload_info -> get_upload_info

taion opened the pull request.

itajaja commented on flask_annex/file.py:
```diff
@@ -92,5 +92,24 @@ def send_file(self, key):
+        )
+
+        upload_info = {
+            'method': 'PUT',
```
shouldn't this be `method`?

taion committed:
Revise upload info support

- send_upload_info -> get_upload_info
- Add max_content_length to S3 annex configuration
- Test fallback upload info with file annex
